### PR TITLE
groff: add recipe for v1.22.4

### PR DESCRIPTION
**Pull request by nehaljwani**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

File: recipes/groff/build.sh [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+# Workaround for long shebang lines
+find $SRC_DIR -type f | \
+    xargs -L1 perl -i.bak \
+        -pe 's,^#!\@PERL\@ -w,#!/usr/bin/env perl,;' \
+        -pe "s,perl -w,perl,;" \
+        -pe "s,$PREFIX/bin/perl,/usr/bin/env perl,;"
+
+# Workarond for dependency issue
+make -j${CPU_COUNT} font/devpdf/build_font_files
+make -j${CPU_COUNT} install
```
Comment (scopatz) [inline recipes/groff/build.sh:15]: Is there a `make check` you can run?

File: recipes/groff/build.sh [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+# Workaround for long shebang lines
+find $SRC_DIR -type f | \
+    xargs -L1 perl -i.bak \
+        -pe 's,^#!\@PERL\@ -w,#!/usr/bin/env perl,;' \
+        -pe "s,perl -w,perl,;" \
+        -pe "s,$PREFIX/bin/perl,/usr/bin/env perl,;"
+
+# Workarond for dependency issue
+make -j${CPU_COUNT} font/devpdf/build_font_files
```
Comment (scopatz) [inline recipes/groff/build.sh:14]: Don't we have fonts elsewhere in conda-forge?

File: recipes/groff/0001-Include-config.h.patch [hunk 3]
```diff
@@ -0,0 +1,1228 @@
...
+index 8db993f..38db396 100644
+--- a/src/preproc/eqn/other.cpp
++++ b/src/preproc/eqn/other.cpp
+@@ -17,6 +17,10 @@ for more details.
+ You should have received a copy of the GNU General Public License
+ along with this program.  If not, see <http://www.gnu.org/licenses/>. */
+ 
++#if HAVE_CONFIG_H
++# include <config.h>
++#endif
```
Comment (scopatz) [inline recipes/groff/0001-Include-config.h.patch:1191]: Why is this needed?

File: recipes/groff/build.sh (hunk 1 above)
Comment (nehaljwani) [reply recipes/groff/build.sh:16]: There is, I added one, but depending on the version of `gs` in `PATH`, it may or may not fail a particular test.

File: recipes/groff/build.sh (hunk 2 above)
Comment (nehaljwani) [reply recipes/groff/build.sh:15]: I have added a comment to describe the actual need for this split make step.

File: recipes/groff/0001-Include-config.h.patch (hunk 3 above)
Comment (nehaljwani) [reply recipes/groff/0001-Include-config.h.patch:1191]: Added another comment to explain the need for that patch.

**Comment by jakirkham** [comment]
@scopatz, would you be able to give this another look? 🙂

File: recipes/groff/0001-Include-config.h.patch (hunk 3 above)
Comment (scopatz) [reply recipes/groff/0001-Include-config.h.patch:1191]: Is this going to go away at some point?

File: recipes/groff/build.sh (hunk 2 above)
Comment (scopatz) [reply recipes/groff/build.sh:15]: Usually we put fonts in separate packages / outputs, though, so then can be reused. Does that not apply here for some reason?

Comment (nehaljwani) [reply recipes/groff/build.sh:15]: All the font related stuff is packaged under <pkg-name>-<pkg-ver> directories: 
```
share/doc/groff-1.22.4/examples
share/groff/1.22.4/font
share/groff/1.22.4/oldfont
share/man/man5/groff_font.5
share/man/man5/lj4_font.5
```

I am not sure if they are meant to be shared. I didn't even find Fedora doing it:
https://fedora.pkgs.org/31/fedora-x86_64/groff-perl-1.22.3-20.fc31.x86_64.rpm.html

File: recipes/groff/0001-Include-config.h.patch (hunk 3 above)
Comment (nehaljwani) [reply recipes/groff/0001-Include-config.h.patch:1191]: To be honest, I am not sure. Not until the upstream maintainers do something about it.

File: recipes/groff/build.sh (hunk 2 above)
Comment (scopatz) [reply recipes/groff/build.sh:15]: Ok thanks!

**Comment by scopatz** [comment]
Thanks!
